mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -1710,6 +1905,432 @@ public override string ToString ()
 		}
 	}
 
+	// Base class for all Sockaddr types.
+	// This class is not abstract, instances of this class can be used to determine the sa_family value.
+	// This class and all classes which are deriving from it and are passed to the native code have to be blittable.
+	[CLSCompliant (false)]
+	[StructLayout (LayoutKind.Sequential)]
+	public class Sockaddr {
+		internal SockaddrType type;
+		public UnixAddressFamily sa_family;
+
+		public Sockaddr ()
+		{
+			this.type = SockaddrType.Sockaddr;
+			this.sa_family = UnixAddressFamily.AF_UNSPEC;
+		}
+		internal Sockaddr (SockaddrType type, UnixAddressFamily sa_family)
+		{
+			this.type = type;
+			this.sa_family = sa_family;
+		}
+
+		[DllImport (Syscall.MPH, SetLastError=true, 
+				EntryPoint="Mono_Posix_Sockaddr_GetNativeSize")]
+		static extern int GetNativeSize (ref _SockaddrDynamic address, out long size);
+		[DllImport (Syscall.MPH, SetLastError=true, 
+				EntryPoint="Mono_Posix_Sockaddr_GetNativeSize")]
+		static extern int GetNativeSize (Sockaddr address, out long size);
+		public unsafe long GetNativeSize ()
+		{
+			long size;
+			if (this != null && this.IsDynamic) {
+				fixed (byte* data = this.DynamicData ()) {
+					_SockaddrDynamic dyn = new _SockaddrDynamic (this, data, false);
```

Comment:
Style: when using boolean parameters, use C#4 named parameters so we know WTF it's controlling:

    new _SockaddrDynamic (this, data, useMaxLength:false);